cmd/ipfs/main.go

```diff
@@ -89,316 +89,132 @@ func mainRet() int {
 	}
 	defer stopFunc() // to be executed as late as possible
 
-	var invoc cmdInvocation
-	defer invoc.close()
-
-	// this is a local helper to print out help text.
-	// there's some considerations that this makes easier.
-	printHelp := func(long bool, w io.Writer) {
-		helpFunc := cli.ShortHelp
-		if long {
-			helpFunc = cli.LongHelp
-		}
-
-		helpFunc("ipfs", Root, invoc.path, w)
-	}
-
-	// this is a message to tell the user how to get the help text
-	printMetaHelp := func(w io.Writer) {
-		cmdPath := strings.Join(invoc.path, " ")
-		fmt.Fprintf(w, "Use 'ipfs %s --help' for information about this command\n", cmdPath)
-	}
+	intrh, ctx := setupInterruptHandler(ctx)
+	defer intrh.Close()
 
 	// Handle `ipfs help'
 	if len(os.Args) == 2 {
 		if os.Args[1] == "help" {
-			printHelp(false, os.Stdout)
-			return 0
+			os.Args[1] = "-h"
```

Comment:
Not new... fix later.

Reply:
Yes, there is. os.Args is simply passed to the cli parser later on. So instead of modifying os.Args we can copy this to a new slice and pass that one to the parser. I changed os.Args because I was afraid something else might later read os.Args and then we'd have inconsistencies.

Reply:
an I actually think it is new

Reply:
Sorry, by not new I mean that we do this elsewhere as well (might as well fix all instances at once).